checksum/crc/crc8.go

```diff
@@ -0,0 +1,53 @@
+// crc8.go
+// description: Calculate CRC8
+// details:
+// A cyclic redundancy check (CRC) is an error-detecting code commonly used in digital networks and storage devices to detect accidental changes to raw data. See more [CRC](https://en.wikipedia.org/wiki/Cyclic_redundancy_check)
+// author(s) [red_byte](https://github.com/i-redbyte)
+// see crc8_test.go
+
+package crc
+
+import "math/bits"
+
+// CRCModel contains the necessary parameters for calculating the DRC algorithm
+type CRCModel struct {
+	Poly   uint8
+	Init   uint8
+	RefIn  bool
+	RefOut bool
+	XorOut uint8
+	Name   string
+}
+
+// CalculateCRC8 This function calculate CRC8 checksum.
+func CalculateCRC8(data []byte, model CRCModel) uint8 {
```

Comment:
Can you split this into smaller functions? This seems overly complicated than the 8-bit arc

Reply:
Done